test-common/test-common-internal-api/src/main/java/org/apache/kafka/common/test/api/ClusterTestThreadLeakIgnore.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.test.api;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Target({METHOD})
+@Retention(RUNTIME)
+public @interface ClusterTestThreadLeakIgnore {
+    // ignore threads left running that have names matching these prefixes
+    // useful when threads are created by client libraries or plugins (e.g. metrics reporters)
```

Comment:
Could we trans this as a javadoc?

Reply:
sure!

Reply:
@frankvicky please note that I did not add the `@Documented` tag as the `DetectThreadLeak` interface is not mentioned in the README. I think javadocs will suffice